packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/BuddyDeployer.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2021, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity ^0.6.11;
+
+import "../ethereum/BuddyBridge.sol";
+
+// TODO: get from arb-os submodule
+interface ArbSys {
+    function sendTxToL1(address destAddr, bytes calldata calldataForL1) external payable;
+}
+
+contract L2Deployer {
+
+    constructor() public {}
+
+    function executeBuddyDeploy(bytes memory deployCode)
+        external
+    {
+        // we don't want nasty address clashes
+        require(msg.sender == tx.origin, "Can't be called by L2 contract");
```

Comment:
We'll want to use `calledFromL1` from https://github.com/OffchainLabs/arb-os/pull/376 to be safe I think